src/scheduler.rs

```diff
@@ -78,10 +92,29 @@ async fn scheduler_loop(
     checker: Arc<impl Checker + 'static>,
     producer: Arc<impl ResultsProducer + 'static>,
     shutdown: CancellationToken,
+    progress_key: String,
+    redis_cluster_nodes: Vec<String>,
 ) {
-    let mut interval = time::interval(time::Duration::from_secs(1));
+    let client = ClusterClient::new(redis_cluster_nodes).unwrap();
+    let mut connection = client
+        .get_async_connection()
+        .await
+        .expect("Unable to connect to Redis");
```

Comment:
Is it OK to do this every time? would it make more sense to pass the connection in when we start the scheduler?

Reply:
oh nvm I see. ignore me